File: 7-kyu/curing-arachnophobia/javascript/solution.js
const drawSpider = (legSize, bodySize, mouth, eye) => {
  const legType = { 1: "^ ^", 2: "/\\ /\\", 3: "/╲ ╱\\", 4: "╱╲ ╱╲" };
  const legs = legType[legSize].split(" ");
  const body = `${"(".repeat(bodySize)} ${")".repeat(bodySize)}`.split(" ");
  const numEyes = 2 ** bodySize;
  const eyes = eye.repeat(numEyes / 2);
  const spider = `${legs[0]}${body[0]}${eyes}${mouth}${eyes}${body[1]}${legs[1]}`;
  return spider;
};

File: 7-kyu/curing-arachnophobia/python/solution.py
def draw_spider(leg_size, body_size, mouth, eye):
    LEGS_LOOKUP = {1: "^ ^", 2: '/\\ /\\', 3: '/╲ ╱\\', 4: '╱╲ ╱╲'}
    legs = LEGS_LOOKUP[leg_size].split()
    body = [f'{"(" * body_size}', f'{")" * body_size}']
    eyes = eye * (1 << (body_size - 1))
    spider = f'{legs[0]}{body[0]}{eyes}{mouth}{eyes}{body[1]}{legs[1]}'
    return spider
